sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -632,6 +634,29 @@ object ScalaJSPluginInternal {
     }
   }
 
+  protected def javaSysProps2JSLib(javaOptions: Seq[String]): VirtualJSFile = {
+    val javaSysPropsPattern = "-D([^=]*)=(.*)".r
+    val javaSysProps = javaOptions.map {
+      case javaSysPropsPattern(propName, propValue) => (propName, propValue)
+    }
```

Comment:
I don't like this. It gives the impression that we support all Java options. Either:

- Add your own key (safe, less compatible)
- Fail if you encounter options we don't support (technically a breaking change, practically not a problem since we forced people to have `fork := false` so `javaOptions` was useless anyways).